### PR TITLE
Make results folder writable (Fix for #2131)

### PR DESCRIPTION
**Pull request by othomann**
# Changes

Make sure the results folder is writable by any user. See #2131.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Make sure the results folder is writable by any user.
```

### CODE REVIEW

**Comment by othomann** [comment]
This solves the problem by adding a new init container that sets the results folder to be writable by all users. It doesn't seem to work when the user is using the podTemplate to set the permissions. @sbwsg can give more details on this.

**Comment by othomann** [comment]
> The code looks good to me. We should probably include a description of this behaviour in `docs/developers/README.md`. Up to you if you want to include that in this PR or I can follow up with another.

Done in the latest commit. All commits were squashed and tests are running.

**Comment by sbwsg** [comment]
I'm adding lgtm in addition to my approve earlier since I think the bug in the linked issue is a bit of a showstopper and the fix is relatively small.  I'd like to have this included in the initial 0.11 release.

/lgtm

File: pkg/pod/pod_test.go [hunk 1]
```diff
@@ -65,6 +65,14 @@ func TestMakePod(t *testing.T) {
 		VolumeMounts: []corev1.VolumeMount{toolsMount},
 	}
 
+	resultsInit := corev1.Container{
```
Comment (vdemeester) [inline pkg/pod/pod_test.go]: So… this is a *kinda **big*** problem on OpenShift as… it fails (running as non-root, …)

Comment (vdemeester) [reply pkg/pod/pod_test.go]: Btw, I don't think this one is actually needed (at least, from the OpenShift point of view, the `emptyDir` should make the path writable)

Comment (sbwsg) [reply pkg/pod/pod_test.go]: Sorry about the extra churn from the chmod.  You're right - emptyDir gets 777 perms: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/emptydir/empty_dir.go#L38-L43

**Pull request merged**